appbuilder/project/project-base.ts

```diff
@@ -101,17 +103,15 @@ export abstract class ProjectBase implements Project.IProjectBase {
 					let appIdentifierInConfigXml = this.getAppIdentifierFromConfigFile(pathToConfigXml, /id\s*=\s*"(\S*)"/).wait();
 
 					if (appIdentifierInAndroidManifest && appIdentifierInConfigXml && appIdentifierInAndroidManifest === appIdentifierInConfigXml) {
-						if (appIdentifierInAndroidManifest === ProjectBase.APP_IDENTIFIER_PLACEHOLDER) {
-							this._platformSpecificAppIdentifier = this.projectData.AppIdentifier;
-						} else {
+						if (appIdentifierInAndroidManifest !== ProjectBase.APP_IDENTIFIER_PLACEHOLDER) {
 							this._platformSpecificAppIdentifier = appIdentifierInAndroidManifest;
 						}
 					} else if (appIdentifierInAndroidManifest || appIdentifierInConfigXml) {
-						this.$errors.failWithoutHelp(`Your package in ${ProjectBase.ANDROID_MANIFEST_NAME} and id in ${ProjectBase.CONFIG_XML_NAME} do not match. They must be the same to be able to build your application.`);
+						let appId = appIdentifierInAndroidManifest || appIdentifierInConfigXml;
```

Comment:
move this above the
```TypeScript
if (appIdentifierInAndroidManifest && appIdentifierInConfigXml && appIdentifierInAndroidManifest === appIdentifierInConfigXml)
```
and change the else if here to:
```TypeScript
else if (appId && (appId !== ProjectBase.APP_IDENTIFIER_PLACEHOLDER))
```
so the else if's body can remain as it was - containing only
```TypeScript
this.$errors.failWithoutHelp(`Your package in ${ProjectBase.ANDROID_MANIFEST_NAME} and id in ${ProjectBase.CONFIG_XML_NAME} do not match. They must be the same to be able to build your application.`);
```